src/libraries/System.Text.Json/gen/JsonSourceGenerator.Emitter.cs

```diff
@@ -1138,9 +1138,19 @@ private string GetLogicForDefaultSerializerOptionsInit()
             {
                 JsonSourceGenerationOptionsAttribute options = _currentContext.GenerationOptions;
 
-                string? namingPolicyInit = options.PropertyNamingPolicy == JsonKnownNamingPolicy.CamelCase
+                string? namingPolicyName = options.PropertyNamingPolicy switch
+                {
+                    JsonKnownNamingPolicy.CamelCase => "CamelCase",
+                    JsonKnownNamingPolicy.SnakeLowerCase => "SnakeLowerCase",
+                    JsonKnownNamingPolicy.SnakeUpperCase => "SnakeUpperCase",
+                    JsonKnownNamingPolicy.KebabLowerCase => "KebabLowerCase",
+                    JsonKnownNamingPolicy.KebabUpperCase => "KebabUpperCase",
+                    _ => null,
+                };
```

Comment:
could this be `string? namingPolicyName = Enum.IsDefined(policy) ? policy.ToString() : null;` ?

Reply:
`IsDefined` cannot be used because under the hood it uses reflection and a binary search. `ToString` isn't fast too, but `nameof` would be more preferrable in my opinion that these hard coded strings.

Reply:
I think that might be fine to use reflection during code-gen - as long as we don't use reflection in the generated code

Reply:
`nameof` is imo the best approach